source/patterns/@aws-solutions-constructs/aws-lambda-opensearch/test/lambda-opensearch.test.ts

```diff
@@ -0,0 +1,583 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { LambdaToOpenSearch, LambdaToOpenSearchProps } from "../lib";
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as cdk from "aws-cdk-lib";
+import '@aws-cdk/assert/jest';
+import * as defaults from '@aws-solutions-constructs/core';
+
+function deployLambdaToOpenSearch(stack: cdk.Stack) {
+  const props: LambdaToOpenSearchProps = {
+    lambdaFunctionProps: getDefaultTestLambdaProps(),
+    openSearchDomainName: 'test-domain'
+  };
+
+  return new LambdaToOpenSearch(stack, 'test-lambda-opensearch-stack', props);
+}
+
+function getDefaultTestLambdaProps(): lambda.FunctionProps {
```

Comment:
Don't need to change this, but it could have been a constant to reduce complexity a little (most compile languages would probably make this optimization at compile time):

```
const defaultTestLambdaProps: lambda.FunctionProps = {
    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
    runtime: lambda.Runtime.NODEJS_14_X,
    handler: 'index.handler',
};
```

Reply:
Agreed, this is better

Reply:
When using this approach, I get an Asset resource already exists in another stack:
`Asset is already associated with another stack 'Default'. Create a new Code instance for every stack.`

Using a function will create a unique reference id for the Code/Asset resource in each CloudFormation stack:
```When the AWS CDK deploys an app that references assets (either directly by the app code or through a library), the AWS CDK CLI first prepares and publishes the assets to an Amazon S3 bucket or Amazon ECR repository, which was created during bootstrapping. Only then are the resources defined in the stack deployed.```